fogros2/fogros2/gcp_cloud_instance.py

```diff
@@ -0,0 +1,150 @@
+
+    def info(self, flush_to_disk=True):
+        info_dict = super().info(flush_to_disk)
+        info_dict["compute_region"] = self.zone
+        info_dict["compute_instance_type"] = self.type
+        info_dict["disk_size"] = self.compute_instance_disk_size
+        info_dict["compute_instance_id"] = self._name
+        if flush_to_disk:
+            with open(os.path.join(self._working_dir, "info"), "w+") as f:
+                json.dump(info_dict, f)
+        return info_dict
+
+    def create_compute_engine_instance(self):
+        os.system(f'gcloud config set project {self._project_id}')
+
+        result = subprocess.check_output(f'gcloud compute instances create {self._name} '
+                                         f'--project={self._project_id} --zone={self.zone} --machine-type={self.type} '
+                                         '--network-interface=network-tier=PREMIUM,subnet=default '
+                                         '--maintenance-policy=MIGRATE --provisioning-model=STANDARD '
+                                         '--scopes=https://www.googleapis.com/auth/devstorage.read_only,'
+                                         'https://www.googleapis.com/auth/logging.write,'
+                                         'https://www.googleapis.com/auth/monitoring.write,'
+                                         'https://www.googleapis.com/auth/servicecontrol,'
+                                         'https://www.googleapis.com/auth/service.management.readonly,'
+                                         'https://www.googleapis.com/auth/trace.append '
+                                         '--create-disk=auto-delete=yes,'
+                                         'boot=yes,'
+                                         f'device-name={self._name},'
+                                         f'image={self.gcp_ami_image},'
+                                         'mode=rw,'
+                                         f'size={self.compute_instance_disk_size},'
+                                         f'type=projects/{self._project_id}/zones/{self.zone}/diskTypes/pd-balanced '
+                                         '--no-shielded-secure-boot '
+                                         '--shielded-vtpm '
+                                         '--shielded-integrity-monitoring '
+                                         '--reservation-affinity=any', shell=True).decode()
+
+        # Grab external IP
+        ip = extract_bash_column(result, 'EXTERNAL_IP')
+
+        # Verifies the response was an ip
+        if len(ip.split('.')) != 4:
+            raise Exception(f'Error creating instance: {ip}')
+
+        self._ip = ip
+
+        # Generate SSH keys
+        os.system(f"printf '\n\n' | gcloud compute ssh {self._name} --zone {self.zone}")
+
+        user = subprocess.check_output('whoami', shell=True).decode().strip()
+
+        # Username
+        self._username = (open(f'/home/{user}/.ssh/google_compute_engine.pub').
```

Comment:
It would be nice to break this up into more separate steps, as it is tough to parse what is going on currently.